block/internal/executing/executor.go

```diff
@@ -281,6 +288,14 @@ func (e *Executor) executionLoop() {
 
 			if err := e.produceBlock(); err != nil {
 				e.logger.Error().Err(err).Msg("failed to produce block")
+				// Check for permanent validation errors that won't resolve on retry
+				if errors.Is(err, errBlockValidation) {
```

Comment:
Technically this is not necessary. We already halt because we do a `sendCriticalError` in the validateBlock failure path.
This will already halt block production.